### PR TITLE
allow 3 hyphens per word

### PR DESCRIPTION
**Pull request by tomlovesgithub**
Word count now allows 3 hyphens per word.
counts every 4th as the start of a new word
'so-no-more-cheating/-can-happen'

[Added suggested wording and styling from the GDS guidelines too
](https://design-system.service.gov.uk/components/character-count/)

### CODE REVIEW

**Comment by HalcyonJAC** [comment]
Hi @tomlovesgithub 

The regex doesn't work well if the words have numbers in them...

...but I am not should whether the system needs to hande this type of input or not...

**Comment by HalcyonJAC** [comment]
Hi @tomlovesgithub 

I think I have found the smallest bug in the world:



....should say '1 word' instead of '1 words'

File: src/components/Form/TextareaInput.vue [hunk 1]
```diff
@@ -26,14 +26,37 @@
...
+        class="govuk-hint govuk-character-count__message"
+      >
+        {{ `You have ${Math.abs(wordsTooMany)} word${Math.abs(wordsTooMany) > 1 ? 's' : ''} remaining` }}
+      </span>
+      <span
+        v-else
+        class="govuk-hint govuk-character-count__message"
+      >
+        {{ `${words.length}/${wordLimit}` }}
+      </span>
```
Comment (warrensearle) [inline src/components/Form/TextareaInput.vue:57]: Could you simplify this with a computed which returns the `wordLimitMessage`?
(don't like seeing so much javascript inline in the template code)

**Review by warrensearle — CHANGES_REQUESTED** [review]

**Comment by warrensearle** [comment]
@tomlovesgithub CircleCI is reporting that four of the word count tests are failing. When I run the tests locally I get the same 4 failing tests. Are you able to fix the tests?

**Review by mbrookeswebdev — APPROVED** [review]

**Review by warrensearle — APPROVED** [review]
👍

**Pull request merged**